metadataserver/auth/auth.go

```diff
@@ -0,0 +1,132 @@
+package auth
+
+import (
+	"crypto"
+	"crypto/rand"
+	"encoding/base64"
+	"encoding/json"
+	"fmt"
+	"strings"
+	"time"
+
+	"github.com/Netflix/titus-executor/metadataserver/identity"
+	jwt "github.com/dgrijalva/jwt-go"
+)
+
+// Authenticator generates and authenticates tokens
+type Authenticator interface {
+	GenerateToken(ttl time.Duration) (string, error)
+	VerifyToken(token string) bool
+}
+
+type envelope struct {
+	// Token token
+	Token string
+
+	// Expiration in nanoseconds
+	Expiration int64
+}
+
+// JWTAuthenticator authenticates using JWT
+type JWTAuthenticator struct {
+	Key []byte
+}
+
+// GenerateToken token
+func (a *JWTAuthenticator) GenerateToken(ttl time.Duration) (string, error) {
+	claims := &jwt.StandardClaims{
+		ExpiresAt: time.Now().Add(ttl).Unix(),
+	}
+	token := jwt.NewWithClaims(jwt.SigningMethodHS256, claims)
+	return token.SignedString(a.Key)
+}
+
+// VerifyToken verifes a token
+func (a *JWTAuthenticator) VerifyToken(token string) (bool, int64) {
+	var claims jwt.StandardClaims
+	jwtToken, err := jwt.ParseWithClaims(token, &claims, func(token *jwt.Token) (interface{}, error) {
+		if _, ok := token.Method.(*jwt.SigningMethodHMAC); !ok {
+			return nil, fmt.Errorf("Unexpected signing method: %v", token.Header["alg"])
+		}
+		return a.Key, nil
+	})
+
+	if err != nil {
+		return false, 0
+	}
+
+	if jwtToken.Valid && jwtToken.Claims.Valid() != nil {
+		return false, 0
+	}
+
+	remaining := time.Until(time.Unix(claims.ExpiresAt, 0))
+	return true, int64(remaining.Seconds())
+}
+
+// CertificateAuthenticator generates and authenticates tokens
+type CertificateAuthenticator struct {
+	Signer *identity.Signer
+}
+
+// GenerateToken generates a token
+func (a *CertificateAuthenticator) GenerateToken(ttl time.Duration) (string, error) {
```

Comment:
Am I missing something here, or isn't most of this functionality in the `jwt-go` library?

Reply:
Or is this being ripped out?

Reply:
It’s being ripped out.

Reply:
:D